src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs

```diff
@@ -0,0 +1,356 @@
+                var methods = ImmutableArray.CreateBuilder<(MethodDeclarationSyntax Syntax, IMethodSymbol Symbol, int Index, Diagnostic? Diagnostic)>();
+                int methodVtableOffset = interfaceContext.MethodStartIndex;
+                foreach (var member in interfaceData.Symbol.GetMembers())
+                {
+                    if (member.Kind == SymbolKind.Method && !member.IsStatic)
+                    {
+                        // We only support methods that are defined in the same partial interface definition as the
+                        // [GeneratedComInterface] attribute.
+                        // This restriction not only makes finding the syntax for a given method cheaper,
+                        // but it also enables us to ensure that we can determine vtable method order easily.
+                        Location? locationInAttributeSyntax = null;
+                        foreach (var location in member.Locations)
+                        {
+                            if (location.SourceTree == interfaceLocation.SourceTree
+                                && interfaceLocation.SourceSpan.Contains(location.SourceSpan))
+                            {
+                                locationInAttributeSyntax = location;
+                            }
+                        }
+
+                        if (locationInAttributeSyntax is null)
+                        {
+                            methods.Add((
+                                null!,
+                                (IMethodSymbol)member,
+                                0,
+                                member.CreateDiagnostic(
+                                    GeneratorDiagnostics.MethodNotDeclaredInAttributedInterface,
+                                    member.ToDisplayString(),
+                                    interfaceData.Symbol.ToDisplayString())));
+                        }
+                        else
+                        {
+                            var syntax = (MethodDeclarationSyntax)interfaceData.Syntax.FindNode(locationInAttributeSyntax.SourceSpan);
+                            var method = (IMethodSymbol)member;
+                            Diagnostic? diagnostic = GetDiagnosticIfInvalidMethodForGeneration(syntax, method);
+                            methods.Add((syntax, method, diagnostic is not null ? methodVtableOffset++ : 0, diagnostic));
+                        }
+                    }
+                }
+                return methods.ToImmutable();
+            });
+
+            // Split the methods we want to generate and the ones we don't into two separate groups.
+            var methodsToGenerate = methodsWithDiagnostics.Where(static data => data.Diagnostic is null);
+            var invalidMethodDiagnostics = methodsWithDiagnostics.Where(static data => data.Diagnostic is not null);
+
+            context.RegisterSourceOutput(invalidTypeDiagnostics, static (context, invalidMethod) =>
```

Comment:
Should this be invalidMethodDiagnostics?

Reply:
```suggestion
            context.RegisterSourceOutput(invalidMethodDiagnostics, static (context, invalidMethod) =>
```